blockchain/stake/staketx_test.go

```diff
@@ -691,43 +674,46 @@ func TestSSGenErrors(t *testing.T) {
 
 	// Verify optional OP_RETURN with no discriminator.
 	var ssgenNoDiscriminator = dcrutil.NewTx(ssgenMsgTxNoDiscriminator)
+	ssgenNoDiscriminator.SetVersion(wire.TxVersionTreasury)
```

Comment:
Agreed that it would have been nice to enforce all votes to be version `3` once the treasury agenda was activated.

[This logic](https://github.com/decred/dcrd/blob/2b9a9cc66ccba1579a290d9d6238dc9473333bff/blockchain/stake/staketx.go#L1122-L1130) enforces that the vote version must be `3` if it contains treasury votes (example [with treasury votes](https://explorer.dcrdata.org/tx/c1aca5b7412cface3ec0ba727a34473a772be1cdf237513d504eb6839e1f6259) and [without](https://explorer.dcrdata.org/tx/04e021cd7842ac9fec1091acdf653b7dfef48da7c3411f0dc7ecb1c28d6bf9a6) in block 620000 on mainnet):
```golang
// If there are votes the TxVersion must be TxVersionTreasury.
// This test is done late in order to allow older versions
// SSGen if there are no votes.
if !(len(votes) > 0 && tx.Version == wire.TxVersionTreasury) {
	str := fmt.Sprintf("SSGen invalid tx version %v",
		tx.Version)
	return nil, stakeRuleError(ErrSSGenInvalidTxVersion,
		str)
}
```

I set the version as `3` in the referenced tests because I interpreted those tests as vote transactions that intend to contain treasury votes but are malformed and therefore error cases.  I see your point that they could also be seen as malformed vote transactions without treasury votes, though, in which case the actual version currently used is `1`.  I'll look at cleaning this up since the version is also wrong in the base `wire.MsgTx`s, as you mentioned.

Reply:
> I set the version as `3` in the referenced tests because I interpreted those tests as vote transactions that intend to contain treasury votes but are malformed and therefore error cases. I see your point that they could also be seen as malformed vote transactions without treasury votes, though, in which case the actual version currently used is `1`. I'll look at cleaning this up since the version is also wrong in the base `wire.MsgTx`s, as you mentioned.

That's a good point.  I was indeed looking at them as malformed votes since they're building on the existing `wire.MsgTx`s, but looking at it again in light of your comment, they do appear to only be used specifically in this context where votes that intend to contain treasury votes are malformed in which case setting them to version `3` as you have does indeed make more sense.

Reply:
I added a commit to set the existing vote `wire.MsgTx`s version to `1`.  I'll also clean up those tests to be a bit easier to follow and cover some of the error cases that are currently missed in a separate PR.

Reply:
Updates look good.